src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+
+  let currentState = { ...state };
+
+  for (let i = 0; i < actions.length; i++) {
+    switch (actions[i].type) {
+      case 'addProperties':
+        currentState = {
+          ...currentState, ...actions[i].extraData,
+        };
```

Comment:
It is better to use [Object.assign](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign) here

Reply:
@polosanya I'm sorry, but in the previous example, the mentor told me that it's better to use the spread operator instead of Object.assign.
There i use Object.assign like this:
case 'addProperties':
        newState = Object.assign({}, prevState, actions[i].extraData);
        
Tell me what better to use - first or second example?